src/content/app/genome-browser/state/focus-object/focusObjectSlice.ts

```diff
@@ -127,20 +131,18 @@ const buildFocusLocationObject = (
 const buildFocusVariantObject = (payload: {
   genomeId: string;
   objectId: string;
-  variantId: string;
+  variant: Variant;
 }): FocusVariant => {
-  const { genomeId, objectId, variantId } = payload;
+  const { genomeId, objectId, variant } = payload;
+  const { name, prediction_results, alleles } = variant;
+
   return {
     type: 'variant',
     genome_id: genomeId,
     object_id: objectId,
-    label: variantId,
-    location: {
-      // just some arbitrary hardcoded location; this will change to proper data in the future
-      chromosome: '13',
-      start: 1_000_000,
-      end: 1_000_001
-    }
+    label: name,
+    prediction_results,
+    alleles
```

Comment:
I suggest reverting all changes to the focus object slice, and instead using `useGbVariantQuery` to fetch all the relevant data, just as `TrackPanelVariant` component does.

Reply:
Hmm I thought we need to fetch the data like we do for the focus gene. What do we do about this then?

https://github.com/Ensembl/ensembl-client/pull/963/files#diff-aa700f886a05d15b66bf85b2c78280b2024596bf76ee209af4f76f1292e2ed22L264

Reply:
✅